c++/src/kj/miniposix.c++

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2013-2014 Sandstorm Development Group, Inc. and contributors
+// Licensed under the MIT License:
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#include "miniposix.h"
+
+namespace kj {
+namespace miniposix {
+#if !defined(_WIN32) && !defined(IOV_MAX) && !defined(UIO_MAX_IOV)
+size_t iovMax() {
+  // Thread-safe & lazily initialized only on first use.
+  static const size_t KJ_IOV_MAX = [] {
+    long iovmax;
+
+    errno = 0;
+    iovmax = sysconf(_SC_IOV_MAX);
+    if (iovmax == -1 && errno) {
+      // assume iovmax == -1 && errno == 0 means "unbounded"
```

Comment:
I just noticed that this logic has changed. The comment is now confusing because it describes a case that is _not_ handled by this block.

It looks like the case that the comment describes is now handled by returning `(size_t) iovmax`, the same code that executes in the case that `sysconf()` returns a positive value.

This looks like it works but only because `-1` is being converted to an unsigned value, and thus effectively becomes `ULONG_MAX`, which effectively expresses "unbounded". But it's pretty confusing to rely on a signed->unsigned conversion just coincidentally producing the right result for two logically distinct cases.

Could you maybe handle this case explicitly and return `kj::maxValue` to make things clearer? (Also make sure to declare the lambda's return type explicitly.)

Reply:
How does it look now?